connector/src/lib.rs

```diff
@@ -100,10 +100,15 @@ pub struct SnapshotSourceConfig {
     pub rpc_http_url: String,
 }
 
+#[derive(Clone, Debug, Deserialize)]
+pub enum EntityFilter {
+    FilterByAccountIds(Vec<String>),
+    FilterByProgramId(String),
+}
+
 #[derive(Clone, Debug, Deserialize)]
 pub struct FilterConfig {
-    pub program_ids: Vec<String>,
-    pub account_ids: Vec<String>,
+    pub entity_filter: EntityFilter,
```

Comment:
this struct is for future-proofing?

Reply:
If we were to use this for mango-v4 (the liquidator would really like to use grpc streams!), we'd need to be able to subscribe to:
- all mango-v4 program accounts
- a list of oracle accounts
- all open book v1 accounts that match a bunch of binary data filters

The idea is that you'd run multiple `source` instances to cover the three streams?

I guess that makes sense, the only share data between them is the slots stream as far as I know.

There may be some added difficulty with waiting for the three inputs to come up and be ready. Maybe later there could be a wrapper for complex setups like that.

Reply:
thx for sharing the use cases.
my changes are guided by the current filter pattern. it might be that websocket+grpc supports the complexity you mentioned in one filter - yet I doubt it.

so yes, we have to support multiple sources with different filters and merge the stream data somehow.

Reply:
updated the initial ticket wrt "multiplexing":
1. websocket+geyser
2. multiple of same type
3. multiple subscriptions(filters)